v4/newrelic/segments.go

```diff
@@ -323,5 +323,15 @@ func (s *ExternalSegment) SetStatusCode(code int) {}
 // NewRoundTripper: You may not need to use StartExternalSegment at all!
 //
 func StartExternalSegment(txn *Transaction, request *http.Request) *ExternalSegment {
-	return nil
+	if nil == txn && nil != request {
+		txn = FromContext(request.Context())
+	}
+	s := &ExternalSegment{
+		StartTime: txn.StartSegmentNow(),
+		Request:   request,
+	}
+
+	txn.InsertDistributedTraceHeaders(request.Header)
```

Comment:
I think we need to check for a nil request and nil headers here. That's what we do in the v3 agent at least.

Reply:
Or at least we should probably have tests for those two cases.

Reply:
Agreed. Fixed and added a test.